components/email_aliases/email_aliases_service.cc

```diff
@@ -30,21 +89,198 @@ void EmailAliasesService::BindInterface(
+        IDS_EMAIL_ALIASES_ERROR_NO_VERIFICATION_TOKEN));
+    return;
+  }
+  // Success; set the verification token and notify observers.
+  verification_token_ = parsed_auth->verification_token;
+  NotifyObserversAuthStateChanged(mojom::AuthenticationStatus::kAuthenticating);
+  std::move(callback).Run(std::nullopt);
+  // Begin the polling window.
+  RequestSession();
+}
+
+void EmailAliasesService::RequestSession() {
+  if (verification_token_.empty()) {
+    // No verification token; polling has been cancelled.
+    return;
+  }
+  SessionRequest session_request;
+  session_request.wait = true;
+  std::optional<std::string> body = base::WriteJson(session_request.ToValue());
+  CHECK(body);
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = GURL(verify_result_url_);
+  resource_request->method = net::HttpRequestHeaders::kPostMethod;
+  resource_request->headers.SetHeader(
+      "Authorization", std::string("Bearer ") + verification_token_);
+  verification_simple_url_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), traffic_annotation);
+  verification_simple_url_loader_->AttachStringForUpload(*body,
+                                                         "application/json");
+  verification_simple_url_loader_->DownloadToString(
+      url_loader_factory_.get(),
+      base::BindOnce(&EmailAliasesService::OnRequestSessionResponse,
+                     weak_factory_.GetWeakPtr()),
+      kMaxResponseLength);
+}
+
+void EmailAliasesService::OnRequestSessionResponse(
+    std::optional<std::string> response_body) {
+  if (!response_body) {
+    // No response body, log it and re-request.
+    LOG(ERROR) << "Email Aliases service error: No response body";
+    MaybeRequestSessionAgain();
+    return;
+  }
+  const auto response_body_dict = base::JSONReader::ReadDict(*response_body);
+  if (!response_body_dict) {
+    // Invalid response body, log it and re-request.
+    LOG(ERROR) << "Email Aliases service error: Invalid response body";
+    MaybeRequestSessionAgain();
+    return;
+  }
+  auto error_message = ErrorResponse::FromValue(*response_body_dict);
+  if (error_message) {
+    // An error has been reported by the server, indicating that verification
+    // failed. Log it and notify observers.
+    LOG(ERROR) << "Email Aliases service error: " << error_message->error;
+    session_poll_elapsed_timer_.reset();
+    NotifyObserversAuthStateChanged(
+        mojom::AuthenticationStatus::kUnauthenticated,
+        /*error_message=*/l10n_util::GetStringUTF8(
+            IDS_EMAIL_ALIASES_ERROR_VERIFICATION_FAILED));
+    return;
+  }
+  auto parsed_session = SessionResponse::FromValue(*response_body_dict);
+  if (!parsed_session) {
+    // No error message but unparseable response, log it and re-request.
+    LOG(ERROR) << "Email Aliases service verification error: Parse error but "
+                  "no error message";
```

Comment:
That will be quite a lot of logs... is that correct?

Reply:
All of these errors indicate a server or connectivity problem. Under normal circumstances, we would not expect to see these log messages.